kernel/thread.c

```diff
@@ -237,6 +237,28 @@ static void _free_zombies(void* arg)
 
 void myst_zombify_thread(myst_thread_t* thread)
 {
+    // If the thread was blocked on a condvar and got killed,
+    // get out of the waiting list of the condvar.
+    // If the thread exits normally, the cond_wait should be NULL.
```

Comment:
Can a thread be killed but was not blocked on a condvar?

Reply:
Yes. In that case, the field `thread->signal.cond_wait` would be NULL.